### PR TITLE
Method select

### PR DESCRIPTION
**Pull request by LuisARodr**
This adds the `select(spec)` method functionality to the framework.
Also includes tests for the method.

Description of the method:
>Creates a new collection containing the elements returned by the spec function.

### CODE REVIEW

File: Methods/ArrayExtension.js [hunk 1]
```diff
@@ -68,12 +68,32 @@ Array.prototype.any = function any(spec) {
...
+ * @throws {TypeError} - If spec is no a function.
+ */
+Array.prototype.select = function select(spec) {
+    if ((typeof spec) != 'function') {
+        throw new TypeError(`Type error: ${spec} is not a function.`);
+    }
 
+    const result = [];
+
+    this.each(function evaluate(value, index) {
```
Comment (jesus-castro3) [inline Methods/ArrayExtension.js:86]: **suggestion**
You can also use reduce here, would come out a bit more cleaner

```
  // remove const result = [];

   return this.reduce((result, value, index) =>  result.push(spec.call(this, value, index)),  []);

   // remove return result;

```

File: Methods/ArrayExtension.js [hunk 2]
```diff
@@ -68,12 +68,32 @@ Array.prototype.any = function any(spec) {
...
 
+/**
+ * Returns a new array with the values returned by the spec function for
+ * each value of the array.
+ * @param {spec} spec - This function is called for every value of the array,
+ * the return value is added to the new array.
+ * @returns {*[]} - A new array with the values returned by spec.
+ * @throws {TypeError} - If spec is no a function.
+ */
+Array.prototype.select = function select(spec) {
```
Comment (jesus-castro3) [inline Methods/ArrayExtension.js:79]: naming your anonymous functions is a good practice, but i don't find much value doing it here again.

File: Methods/ArrayExtension.js (hunk 1 above)
Comment (LuisARodr) [reply Methods/ArrayExtension.js:86]: Tried using reduce as this:
``` javascript
return this.reduce((result, value, index) =>  {
  result.push(spec.call(this, value, index));
  return result;
},  []);
```
but found a problem where it doesn't work for an empty array eg. `new Array(10)`, and that caused the first test to fail, did some testing and noticed that other Array functions like 'forEach' also don't work with empty arrays from `new Array(n)`.

File: Methods/ArrayExtension.js (hunk 2 above)
Comment (LuisARodr) [reply Methods/ArrayExtension.js:79]: Then, the problem is that I'm assigning the `select` function to `prototype.select` but inside the function I'm not making use of the name?

**Comment by LuisARodr** [comment]
Getting closed to merge into [branch dev](https://github.com/LuisARodr/array-extension/tree/dev)

**Pull request closed without merging**